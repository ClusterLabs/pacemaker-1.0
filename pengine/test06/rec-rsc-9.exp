 <transition_graph cluster-delay="60s" stonith-timeout="60s" failed-stop-offset="INFINITY" failed-start-offset="INFINITY" batch-limit="30" transition_id="0">
   <synapse id="0">
     <action_set>
       <rsc_op id="5" operation="start" operation_key="rsc1_start_0" on_node="node1" on_node_uuid="node1">
         <primitive id="rsc1" long-id="rsc1" class="heartbeat" type="apache"/>
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="6" operation="stop" operation_key="rsc1_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="15" operation="running" operation_key="bar_running_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="1">
     <action_set>
       <rsc_op id="6" operation="stop" operation_key="rsc1_stop_0" on_node="node1" on_node_uuid="node1">
         <primitive id="rsc1" long-id="rsc1" class="heartbeat" type="apache"/>
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="2">
     <action_set>
       <pseudo_event id="9" operation="start" operation_key="foo_start_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <pseudo_event id="11" operation="stop" operation_key="foo_stop_0"/>
       </trigger>
       <trigger>
         <pseudo_event id="12" operation="stopped" operation_key="foo_stopped_0"/>
       </trigger>
       <trigger>
         <pseudo_event id="15" operation="running" operation_key="bar_running_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="3">
     <action_set>
       <pseudo_event id="10" operation="running" operation_key="foo_running_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="7" operation="start" operation_key="rsc2_start_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="9" operation="start" operation_key="foo_start_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="4">
     <action_set>
       <pseudo_event id="11" operation="stop" operation_key="foo_stop_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="5">
     <action_set>
       <pseudo_event id="12" operation="stopped" operation_key="foo_stopped_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="8" operation="stop" operation_key="rsc2_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="11" operation="stop" operation_key="foo_stop_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="6">
     <action_set>
       <rsc_op id="7" operation="start" operation_key="rsc2_start_0" on_node="node1" on_node_uuid="node1">
         <primitive id="rsc2" long-id="foo:rsc2" class="heartbeat" type="apache"/>
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="8" operation="stop" operation_key="rsc2_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="9" operation="start" operation_key="foo_start_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="7">
     <action_set>
       <rsc_op id="8" operation="stop" operation_key="rsc2_stop_0" on_node="node1" on_node_uuid="node1">
         <primitive id="rsc2" long-id="foo:rsc2" class="heartbeat" type="apache"/>
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <pseudo_event id="11" operation="stop" operation_key="foo_stop_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="8">
     <action_set>
       <pseudo_event id="14" operation="start" operation_key="bar_start_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <pseudo_event id="16" operation="stop" operation_key="bar_stop_0"/>
       </trigger>
       <trigger>
         <pseudo_event id="17" operation="stopped" operation_key="bar_stopped_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="9">
     <action_set>
       <pseudo_event id="15" operation="running" operation_key="bar_running_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="13" operation="start" operation_key="rsc3_start_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="14" operation="start" operation_key="bar_start_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="10">
     <action_set>
       <pseudo_event id="16" operation="stop" operation_key="bar_stop_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="6" operation="stop" operation_key="rsc1_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="12" operation="stopped" operation_key="foo_stopped_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="11">
     <action_set>
       <pseudo_event id="17" operation="stopped" operation_key="bar_stopped_0">
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="1" operation="stop" operation_key="rsc3_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="16" operation="stop" operation_key="bar_stop_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="12">
     <action_set>
       <rsc_op id="1" operation="stop" operation_key="rsc3_stop_0" on_node="node1" on_node_uuid="node1">
         <primitive id="rsc3" long-id="bar:rsc3" class="heartbeat" type="apache"/>
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <pseudo_event id="16" operation="stop" operation_key="bar_stop_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="13">
     <action_set>
       <rsc_op id="13" operation="start" operation_key="rsc3_start_0" on_node="node1" on_node_uuid="node1">
         <primitive id="rsc3" long-id="bar:rsc3" class="heartbeat" type="apache"/>
         <attributes CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="1" operation="stop" operation_key="rsc3_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <pseudo_event id="14" operation="start" operation_key="bar_start_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="14">
     <action_set>
       <pseudo_event id="2" operation="all_stopped" operation_key="all_stopped">
         <attributes crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="1" operation="stop" operation_key="rsc3_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <rsc_op id="6" operation="stop" operation_key="rsc1_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
       <trigger>
         <rsc_op id="8" operation="stop" operation_key="rsc2_stop_0" on_node="node1" on_node_uuid="node1"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="15" priority="1000000">
     <action_set>
       <rsc_op id="4" operation="probe_complete" operation_key="probe_complete" on_node="node1" on_node_uuid="node1">
         <attributes CRM_meta_op_no_wait="true" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs/>
   </synapse>
 </transition_graph>

