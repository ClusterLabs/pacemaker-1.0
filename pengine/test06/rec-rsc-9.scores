Allocation scores:
native_color: rsc1 allocation score on node1: 100
group_color: foo allocation score on node1: 0
group_color: rsc2 allocation score on node1: 100
native_color: rsc2 allocation score on node1: 100
group_color: bar allocation score on node1: 0
group_color: rsc3 allocation score on node1: 100
native_color: rsc3 allocation score on node1: 100
