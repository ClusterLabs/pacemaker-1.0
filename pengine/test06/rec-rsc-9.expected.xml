<?xml version="1.0" encoding="UTF-8"?>
<cib admin_epoch="0" epoch="1" num_updates="1" dc-uuid="0" have-quorum="false" remote-tls-port="0" validate-with="pacemaker-1.0">
  <configuration>
    <crm_config><cluster_property_set id="cib-bootstrap-options"><nvpair id="nvpair" name="no-quorum-policy" value="ignore"/><nvpair id="nvpair" name="symmetric-cluster" value="true"/><nvpair id="nvpair" name="default-resource-stickiness" value="100"/></cluster_property_set></crm_config>
    <nodes>
      <node id="node1" uname="node1" type="member"/>
    </nodes>
    <resources>
      <primitive id="rsc1" class="heartbeat" type="apache"><meta_attributes id="primitive-rsc1.meta"><nvpair id="restart_type.meta.auto-23" name="restart-type" value="restart"/></meta_attributes></primitive>
      <group id="foo"><meta_attributes id="group-foo.meta"><nvpair id="restart_type.meta.auto-25" name="restart-type" value="restart"/></meta_attributes>
	<meta_attributes id="meta_attributes"><nvpair id="nvpair" name="restart-type" value="restart"/></meta_attributes>
	<primitive id="rsc2" class="heartbeat" type="apache"><meta_attributes id="primitive-rsc2.meta"/></primitive>
      </group>
      <group id="bar"><meta_attributes id="group-bar.meta"/>
	<primitive id="rsc3" class="heartbeat" type="apache"><meta_attributes id="primitive-rsc3.meta"/></primitive>
      </group>
    </resources>
    <constraints>
      <rsc_order id="order_recover-1" first="bar" then="rsc1" then-action="start" first-action="start" score="0"/>
      <rsc_order id="order_recover-2" first="bar" then="foo" then-action="start" first-action="start" score="0"/>
    </constraints>
  </configuration>
  <status>
    <node_state id="node1" ha="active" uname="node1" crmd="online" join="member" expected="member" in_ccm="true">
      <lrm id="any">
        <lrm_resources id="lrm_resources">
          <lrm_resource id="rsc1" class="heartbeat" type="apache">
            <lrm_rsc_op id="rsc1_start_0" operation="start" interval="0" op-status="0" rc-code="0" call-id="1" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8" crm_feature_set="1.0.6" transition-magic=""/>
          </lrm_resource>
          <lrm_resource id="rsc3" class="heartbeat" type="apache">
            <lrm_rsc_op id="rsc3_start_0" operation="start" interval="0" op-status="0" rc-code="0" call-id="1" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8" crm_feature_set="1.0.6" transition-magic=""/>
            <lrm_rsc_op id="rsc3_fail_0" operation="fail" interval="0" op-status="4" rc-code="60" call-id="2" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8" crm_feature_set="1.0.6" transition-magic=""/>
          </lrm_resource>
          <lrm_resource id="rsc2" class="heartbeat" type="apache">
            <lrm_rsc_op id="rsc2_start_0" operation="start" interval="0" op-status="0" rc-code="0" call-id="1" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8" crm_feature_set="1.0.6" transition-magic=""/>
          </lrm_resource>
        </lrm_resources>
      </lrm>
    </node_state>
  </status>
</cib>
